src/lib/validate-types.ts

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The Pepr Authors
+import { GroupVersionKind, KubernetesObject } from "kubernetes-fluent-client";
+import { Operation } from "./mutate-types";
+
+/**
+ * GroupVersionResource unambiguously identifies a resource. It doesn't anonymously include GroupVersion
+ * to avoid automatic coercion. It doesn't use a GroupVersion to avoid custom marshalling
+ */
+export interface GroupVersionResource {
+  readonly group: string;
+  readonly version: string;
+  readonly resource: string;
+}
```

Comment:
Do generic kinds needed by all the app need to go to a validate types? This is a little confusing to me. I see mutate processor pulling from validate-types too

Reply:
I had moved them initially to fix the circular dependency. But now that I've made some other changes, I'll go back to revisit the initial changes to see if I can move things back to types.ts so that both mutate and validate can import them from there.

Reply:
bloating our softwares size is going to cause even more memory problems, this is not the approach we should take when fixing circular dependencies because we are not introducing more problems. We should fix the code that was already merged in that duplicated types